### ISSUE TITLE
Document changes in DotNetTool Module starting with Cake 1.1.0

### ISSUE THREAD

**Issue by augustoproiete** [cake-build/website#1739]
### Recommendation

- If you use `Cake.DotNetTool.Module` already on your builds, you should either:

- (1) Upgrade Cake to version 1.1.0, and _remove_ the `Cake.DotNetTool.Module` from your build script (as it's no longer needed)

or

- (2) Pin `Cake.DotNetTool.Module` to a version compatible with the Cake version that you use. `Cake.DotNetTool.Module` version 1.0.1 is the last version compatible with Cake 1.0.0. Starting with this release, `Cake.DotNetTool.Module` will always be released together with Cake and will only be compatible with the current release.

**Comment by TomerPacific** [comment]
Hi @augustoproiete, I'd like to help out. What are the changes you would like done exactly?

**Comment by augustoproiete** [comment]
Hi @TomerPacific thanks for your interest in this one.

The intent is to add the recommendation above to this page here: https://cakebuild.net/extensions/cake-dotnettool-module/

**Comment by TomerPacific** [comment]
@augustoproiete, I've taken a look inside the project to see how I can add what you have written and am coming up short.
I found inside **extensions** folder the file for the DotNetToolModule, but that is a yml file and I don't think I am supposed to add that there (based on looking at other extensions as well).

I've been looking at the [7zip extension](https://cakebuild.net/extensions/cake-7zip/) as a go to guide to follow, but have yet to find out how the Aliases text has been added there. From a brief search I found out the **config.wyam** file, but I am not sure how it all fits in together. I would appreciate it if you could declutter some of this.

**Comment by augustoproiete** [comment]
@TomerPacific Thanks for looking deeper into this one.

We [discussed](https://github.com/cake-build/website/pull/1743) this a while ago, and I believe the easiest path forward is to add a hard-coded alert in the page for now (e.g. `if (extensionName == "Cake.DotNetTool.Module") @Html.Partial("_DotNetToolModuleDescription"); ...` given that we don't yet have a feature that allows us to add custom notes about a specific extension.

![image](https://user-images.githubusercontent.com/177608/135932961-005353d5-390c-48c4-95f7-e4161e7fc926.png)

As part of this change, we should also rollback the last changes that were made to `AnalyzedPackageVersion` and `AnalyzedPackagePublishDate` to reflect the latest "valid" version of this extension, as per @pascalberger's [comments here](https://github.com/cake-build/website/pull/1743#issuecomment-792333452):

```diff
-AnalyzedPackageVersion: 1.1.0
+AnalyzedPackageVersion: 1.0.1
AnalyzedPackageIsPrerelease: false
-AnalyzedPackagePublishDate: 2021-03-06T06:37:35.9600000Z
+AnalyzedPackagePublishDate: 2021-02-22T20:53:19.0970000Z
```

---

The description you see at the top is coming from the `description` field inside the [`Cake.DotNetTool.Module.yml`](https://github.com/cake-build/website/blob/master/extensions/Cake.DotNetTool.Module.yml#L9) which normally would be automatically updated from the NuGet package description metadata through a tool we use that keeps these `.yml` files in sync ([AddinDiscoverer](https://github.com/cake-contrib/Cake.AddinDiscoverer)). However [we disabled automatic updates when we shipped Cake v1.1.0](https://github.com/cake-contrib/Cake.AddinDiscoverer/blob/3.78/Source/Cake.AddinDiscoverer/exclusionlist.json#L19) so updating the `description` field in the `.yml` would be a possible route, but that only supports simple/raw text i.e. Doesn't allow us to add HTML or Markdown (for the alert).

**Comment by TomerPacific** [comment]
thanks for the lengthy explanation, @augustoproiete. So, if I understand correctly, since the documentation field in the yml file does not support markup, adding the required text inside of an alert will take place inside the **__ExtensionsPage.cshtml_**?

**Comment by augustoproiete** [comment]
> adding the required text inside of an alert will take place inside the _ExtensionsPage.cshtml?

@TomerPacific Yes , except that to make it a little cleaner, lets store the alert text on a separate file e.g. `_ExtensionDotNetToolModule.cshtml` and modify `_ExtensionsPage.cshtml` to render this file/partial view only for the `Cake.DotNetTool.Module`

**Comment by TomerPacific** [comment]
@augustoproiete, I have started working on this and would like to get your input on what I am doing.

- Looking over `_ExtensionsPage.cshtml`, I see that it is using `_ExtensionsList.cshtml` to render all the extensions. Shouldn't I add the condition inside of _ExtensionsList when the anchor tag is clicked? I am having trouble understanding where the data is rendered.
- I have reverted back the version and publish date of the dotnet module
- I have created a new file, `_ExtensionDotNetToolModule.cshtml`
- To add the alert, I looked at `_AddinAlert.cshtml` ass an example
- Is there a source I could look at to better understand the syntax and code you are using? I looked a bit inside the Wyam website, but am looking for more concrete examples

**Comment by augustoproiete** [comment]
@TomerPacific The description of the extension that you see on the [individual page](https://cakebuild.net/extensions/cake-dotnettool-module/) of each extension is rendered in [_ExtensionsLayout.cshtml](https://github.com/cake-build/website/blob/master/input/_ExtensionsLayout.cshtml#L32). This is where you would add the condition to render the partial view with the alert.

> Is there a source I could look at to better understand the syntax and code you are using?

I'm not sure what kind of syntax you're referring to, but all `.cshtml` files are using the [Razor syntax](https://aspnetcore.readthedocs.io/en/stable/mvc/views/razor.html) and the alerts are from [Bootstrap v3.3](https://getbootstrap.com/docs/3.3/components/#alerts), if that helps.

**Comment by TomerPacific** [comment]
@augustoproiete, how's it looking so far?

![Cake](https://user-images.githubusercontent.com/23402988/136712578-db613310-0ec8-49f0-a0ad-4ec339f77d9c.jpg)
